From c585f9f0da806d6256eab76c1dbaa7102062aced Mon Sep 17 00:00:00 2001
From: Piotr Luc <piotr.luc@intel.com>
Date: Wed, 4 Jan 2017 14:29:30 +0100
Subject: [PATCH] x86/mwait-idle: add Knights Mill CPUID

Add Knights Mill (KNM) to the list of CPUIDs supported by mwait-idle.

Signed-off-by: Piotr Luc <piotr.luc@intel.com>
Signed-off-by: Len Brown <len.brown@intel.com>
[Linux commit: a2c1bc645e87346150516b3abf1933ed29d0f48b]
Signed-off-by: Jan Beulich <jbeulich@suse.com>
Acked-by: Andrew Cooper <andrew.cooper3@citrix.com>
diff --git a/xen/arch/x86/cpu/mwait-idle.c b/xen/arch/x86/cpu/mwait-idle.c
index 9da43847cf..53b5265f46 100644
--- a/xen/arch/x86/cpu/mwait-idle.c
+++ b/xen/arch/x86/cpu/mwait-idle.c
@@ -953,6 +953,7 @@ static const struct x86_cpu_id intel_idle_ids[] __initconstrel = {
 	ICPU(0x9e, skl),
 	ICPU(0x55, skx),
 	ICPU(0x57, knl),
+	ICPU(0x85, knl),
 	ICPU(0x5c, bxt),
 	ICPU(0x5f, dnv),
 	{}
